crates/hyperqueue/src/client/commands/job.rs

```diff
@@ -1,13 +1,52 @@
 use crate::client::globalsettings::GlobalSettings;
 use crate::client::job::get_worker_map;
+use crate::client::output::outputs::OutputStream;
 use crate::client::status::{job_status, Status};
+use crate::common::arraydef::IntArray;
+use crate::common::cli::SelectorArg;
 use crate::rpc_call;
 use crate::transfer::connection::ClientConnection;
 use crate::transfer::messages::{
     CancelJobResponse, CancelRequest, FromClientMessage, JobDetailRequest, JobInfoRequest,
     Selector, ToClientMessage,
 };
 use crate::JobId;
+use clap::Parser;
+
+#[derive(Parser)]
+pub struct JobListOpts {
+    pub job_filters: Vec<Status>,
+}
+
+#[derive(Parser)]
+pub struct JobInfoOpts {
+    /// Single ID, ID range or `last` to display the most recently submitted job
+    pub selector_arg: SelectorArg,
+
+    /// Include detailed task information in the output
+    #[clap(long)]
+    pub tasks: bool,
+}
+
+#[derive(Parser)]
+pub struct JobCancelOpts {
+    /// Select job(s) to cancel
+    pub selector_arg: SelectorArg,
+}
+
+#[derive(Parser)]
+pub struct JobCatOpts {
+    /// Select job
+    pub job_id: u32,
+
+    /// Select task(s) outputs to view
+    #[clap(long)]
+    pub tasks: SelectorArg,
```

Comment:
This should be optional (if not specified, all tasks should be outputted).
`pub tasks: Option<SelectorArg>`